### ISSUE TITLE
Office.context.mailbox.item.requiredAttendees.getAsync returns undefined attendees

### ISSUE THREAD

**Issue by Enotsoul** [OfficeDev/office-js#1170]
We have an issue which first started manifesting itself in January on our staging servers.
**Office.context.mailbox.item.requiredAttendees.getAsync** returns undefined attendees even though there are attendees!
All attendees are fully resolved (image+name) in both outlook and OWA.

Gradually the same behaviour expanded to our test and even to production rendering our app useless
Needless to say, the expected behaviour functioned for years with the same code.

## Expected Behavior
Office.context.mailbox.item.requiredAttendees.getAsync(handleRequiredAttendees);

It would be expected that we received a list of all required attendees would contain the e-mail address and name.

if we have 5 required attendees (2 people and 3 rooms) we expect to get
```
{
  "value": [
	{
	  "emailAddress": "Sarah@example.onmicrosoft.com",
	  "displayName": "Sarah",
	  "recipientType": "other"
	},
... other people
	{
	  "emailAddress":  "Excalibur.room@example.onmicrosoft.com"
	  "displayName": "Excalibur MeetingRoom",
}
... other rooms
  ],
  "status": "succeeded"
}

```
## Current Behavior
Currently we have 2 very weird behaviours.
In OWA we sometimes get to resolve the Room attendees. In OUtlook we fail to get users. 

```
{
  "value": [
	{
	  "emailAddress": "Sarah@example.onmicrosoft.com",
	  "displayName": "Sarah",
	  "recipientType": "other"
	},
... other people
	{
	  "emailAddress":  undefined
	  "displayName": undefined
}
... other rooms
  ],
  "status": "succeeded"
}

```
Depending on the context of OWA/outlook we get rooms as undefined or the current user as undefined.. or even other attendees.

## Steps to Reproduce, or Live Example
Deploy an OUtlook Plugin with the rights to be able to read and write data to the mailbox.	
```
var app = Office.context.mailbox;
  app.item.requiredAttendees.getAsync(function (asyncResult) {
    
                console.log(" required attendees ", asyncResult) // Data is already missing here!
                if (asyncResult.status === Office.AsyncResultStatus.Failed) {
                    deferred.reject(asyncResult.error);
                } else {
                    requiredAttendees = asyncResult.value;
                    app.item.location.getAsync(function (asyncResult) {
                        if (asyncResult.status === Office.AsyncResultStatus.Failed) {
                            console.error(asyncResult.error);
                            Raven.captureMessage("Couldn't retrieve location.");
                            deferred.reject(asyncResult.error);
                        } else {

                            addMissingRoomsFromLocationString(asyncResult.value, requiredAttendees);

                            deferred.resolve({
                                status: asyncResult.status,
                                value: requiredAttendees
                            });
                        }
                    });
                }
            }
```

It's as simple as the above snippet, howver if we can't trust the data because office.js returns undefined the whole point of our app becomes unusable.

## Context

This was something which worked in the past, it rendered us unable to use the plugin and our customers are thus unable to do anything since this basic feature is broken! We expected this to be solved in at least 1-2 weeks, however it's been going on for a few months.

Receiving data as "undefined" crashes our app down the line and we are unable to do anything within the app since this basic functionality is missing...

## Your Environment
We've tested on multiple devices including multiple versions of outlook. ON various Office 365 environments.
We tried various expriments even going back to a previous version of our codebases  which resulted in the same issues.

Even by logging in with multiple mailboxes!
* Platform:  Office Online, PC Desktop Mac
* Host: Outlook
* Office version number: Outlook 365 16.0.12740.20144 32 bit
* Operating System: Windows 10
* Browser for OWA: Firefox(latest), Chrome (latest), Edge (latest) 

## Useful logs

![image](https://user-images.githubusercontent.com/19469665/82345048-79f38700-99fd-11ea-8a79-7e85f7e575d0.png)
![image](https://user-images.githubusercontent.com/19469665/82345146-95f72880-99fd-11ea-8895-0885e963ac03.png)

**Comment by exextoc** [comment]
We are looking into this issue to try to get an in-house repro.

**Comment by exextoc** [comment]
We are still unable to reproduce this issue in-house. Looking at your first screenshot, at the bottom right section, it appears you are creating a meeting in a Shared Folder / Calendar. If that is correct, does this issue only occur when using a shared calendar, or does it occur in non-shared calendars as well?

**Comment by Enotsoul** [comment]
Hello,

The shared folder/calendar is irrelevant as this occurs with a variety of users and on different environments even with non-shared calendars. 

Even if it had something to to with shared folders I doubt this would be the problem since it had worked well in the past.

**Comment by exextoc** [comment]
We are unable to repro this issue. Does this issue repro in other add-ins? 

Another add-in to use for testing can be **ScriptLab for Outlook**, which is available in the store. Instructions for installing add-ins from the store can be found [here](https://support.office.com/en-us/article/get-an-office-add-in-for-outlook-1ee261f9-49bf-4ba6-b3e2-2ba7bcab64c8#:~:text=Install%20an%20add%2Din%20from,want%2C%20review%20its%20privacy%20polic).

In this add-in, there are sample snippets, such as **Get and set required attendees (Appointment Organizer)**, that can be used to try to repro this issue in another add-in.

**Comment by Enotsoul** [comment]
Unfortunately, other addins don't make usage of the whole functionality which we use, and neither could I find an addin which adds rooms AND users to test this out.

Will look more into this, however it seems that that specific functionality doesn't work properly on the API side

**Comment by exextoc** [comment]
@Enotsoul - ScriptLab's playground enables you to emulate your behavior using the APIs from sample snippets. You can use those snippets to set and get your attendees. For users, you'll use the setAttendees API passing in the details of the user, then you can use the enahancedLocation APIs to set conference rooms and lastly, make a call to getRequiredAttendees to see if you still see this behavior.

**Comment by Enotsoul** [comment]
Greetings once again,

I've installed ScriptLab for OUtlook and tested it in both Outlook (for Windows) and OWA on multiple users and environments. Just to be sure we could replicate things evenly I took my time in testing a variety of things with the snippets provided.

I've used the set and get required/optional attendees snippets provided by ScriptLab UNALTERED. 

Outlook:
When getting the required attendees the system only returns the users email addresses (including current user), rooms and other e-mail resources are not returned. 
OWA:
Returns everything except the current user
_Just so you know, ScriptLab is not working in Firefox, I've only managed to get it working in Edge._
Getting the organizer only results in UNDEFINED.

NOTE: I've also tested with optional attendees in Outlook.. and get the same results.

The conclusion:
The problem is VERY REAL and replicates itself with ScriptLab aswell. 
In a real world example, a user would open an existing e-mail/meeting where there are probably 10 different resources and/or users already added. If the code doesn't return the correct info.. it's not usable.

Here's a screenshot from OUtlook:
![image](https://user-images.githubusercontent.com/19469665/84745358-1f7a2600-afbd-11ea-9218-c3828d2aeffb.png)

The following is not a critic it's sincere user feedback
We've pondered and experimented many different things, and we're still blocked by the fact that OfficeJS will NOT return the correct data!
Getting the location string is not a solution since the name could differ from the email address in a variety of complex ways for our customers.

Our current codebase is already filled with various checks and switches between OWA/Outlook code and even checks to see if we're in MAC. All these checks together with expensive implementations to avoid discrepancies and/or problems which exist due to the differences in OWA/Outlook/MAC environments create a great complexity for something so simple as an API. 

The dev time and/or testing time it took us to develop this plugin by going around and solving all the compatibility issues (which should be done at an API not end user!) could realistically amount to just reinvent the wheel.

Do you have anyother ideas or things we could do? Ideas of alternative ways in which this would work?

Or better yet, a real working solution for this issue? 


Thanks!

**Comment by danielebercovici** [comment]
Thanks for the feedback and for reporting this issue regarding discrepancy between OWA and desktop. It has been put on our backlog. We unfortunately have no timelines to share at this point

To get rooms/locations you can use [enhancedLocation](https://docs.microsoft.com/en-us/office/dev/add-ins/outlook/get-or-set-the-location-of-an-appointment).

**Comment by exextoc** [comment]
Thanks for the feedback and for reporting this issue regarding discrepancy between OWA and desktop. It has been put on our backlog. We unfortunately have no timelines to share at this point

To get rooms/locations you can use [enhancedLocation](https://docs.microsoft.com/en-us/office/dev/add-ins/outlook/get-or-set-the-location-of-an-appointment).

**Comment by Enotsoul** [comment]
Hi there,

THanks for the reference to enhancedLocation. We've been able to implement a workaround which seems to work in Outlook. Still a  small issue exists on OWA. 

It will still require further testing locally after which we'll launch it on our staging and production environments to validate that this does indeed provide a long awaited workaround/fix.

Will keep you posted.

**Comment by exextoc** [comment]
Do you have more info about the small issue in OWA? Or is that an issue to be discussed in a new post?
